### PR TITLE
feat: seeding in development

### LINKED ISSUE #11: Script for adding dummy data
- write a script which will create dummy users, groups and chats
- As when locally deployed initially no users present to add to the group
image

### PR DESCRIPTION
**Pull request by priyanshuverma-dev**
## 📋 Description

This PR adds a seed file to seed dummy data in development

## 🔨 Changes Made

- adds a `seed.js` file to seed dummy data in database for development
- changes guest credentials to one of the seeded user
- updates `README.md` about seed command

## ✅ Checklist

Before submitting the PR, please make sure you have completed the following:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation (if applicable).
- [x] My changes generate no new warnings.
- [x] I have tested it locally and it works fine.
- [x] Any dependent changes have been merged and published in downstream modules.

## 🏷️ Types of Changes

What type of changes does your code introduce? (Check all that apply)

- [ ] Bug fix (non-breaking change which fixes an issue) 🐛
- [x] New feature (non-breaking change which adds functionality) ✨
- [ ] UI enhancement (non-breaking change which enhances UI) 🎨
- [x] Documentation update 📚

## 🤝 Related Issues

- Fixes #11

## 📸 Screenshots (if applicable)

---
image

## Reminder:
Please ensure your pull request is made to the `test` branch, not `main`.

*Thank you for contributing to VigneshDevHub! We look forward to reviewing your PR and getting it merged!*

### CODE REVIEW

File: backend/scripts/seed.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+// Seed the database
+const seedDatabase = async () => {
+  await connectDB();
+
+  try {
+    // Clean up the existing data
+    await User.deleteMany();
+    await Chat.deleteMany();
+    await Message.deleteMany();
```
Comment (kartikayasijaa) [inline backend/scripts/seed.js:87]: Don't delete existing data

File: backend/scripts/seed.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    messages.push(newMessage);
+  }
+
+  await Message.insertMany(messages);
+  console.log(`${numMessages} dummy messages created!`);
+};
+
+// Seed the database
+const seedDatabase = async () => {
+  await connectDB();
```
Comment (kartikayasijaa) [inline backend/scripts/seed.js:81]: Can you move this `connectDB()` under try?

File: backend/scripts/seed.js (hunk 1 above)
Comment (priyanshuverma-dev) [reply backend/scripts/seed.js:87]: Ok

File: backend/controllers/userControllers.js [hunk 3]
```diff
@@ -17,7 +17,7 @@ const allUsers = asyncHandler(async (req, res) => {
 });
 
 const registerUser = asyncHandler(async (req, res) => {
-  console.log('hi')
+  console.log("hi");
```
Comment (kartikayasijaa) [inline backend/controllers/userControllers.js:20]: delete this console.log

File: backend/controllers/userControllers.js [hunk 4]
```diff
@@ -58,7 +58,7 @@ const authUser = asyncHandler(async (req, res) => {
   const { email, password } = req.body;
 
   const user = await User.findOne({ email });
-  console.log(generateToken(user._id))
+  console.log(generateToken(user._id));
```
Comment (kartikayasijaa) [inline backend/controllers/userControllers.js:61]: delete this as well

**Comment by kartikayasijaa** [comment]
Bro, added some comments. You did a great job 🥇

**Comment by priyanshuverma-dev** [comment]
@kartikayasijaa sir see the changes are done

**Review by priyanshuverma-dev — COMMENTED** [review]
Changes are done as required

**Comment by priyanshuverma-dev** [comment]
Hi @Sibam-Paul,
Please review the PR.

**Review by Sibam-Paul — APPROVED** [review]

**Pull request merged**